cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -551,7 +587,16 @@ struct check_datetime_format {
     if (d_strings.is_null(idx)) return false;
     string_view d_str = d_strings.element<string_view>(idx);
     if (d_str.empty()) return false;
-    return check_string(d_str);
+    int32_t dateparts[] = {0, 0, 0};  // year, month, day
+    if (!check_string(d_str, dateparts)) return false;
+    auto year  = dateparts[TP_YEAR];
+    auto month = static_cast<uint32_t>(dateparts[TP_MONTH]);
+    auto day   = static_cast<uint32_t>(dateparts[TP_DAY]);
+    if (year == 0 || month == 0 || day == 0) return true;
```

Comment:
What's this extra check for?

Reply:
There are cases where only a single specifier is valid like `%d` for just days.

Reply:
Actually, I found a better way to handle this so I can remove this check.